### DISCUSSION TITLE
using new

### DISCUSSION THREAD

**Discussion by SupinePandora43** [dotnet/csharplang#6103] [Language Ideas] [answered]
# Example

```cs
public class A : IDisposable {
    public A(){}
    public void Dispose(){}
}
```
```cs
static void Test(A a){}
Test(using new A());
```
```cs
public ref struct FuncTime {
    private string? name;
    private Stopwatch sw = Stopwatch.StartNew(); // use global time instead.
    public FuncTime([CallerMemberName] string? name = null){ this.name = name; }
    public void Dispose(){
        if(name is not null) Console.WriteLine($"{name} - {sw.ElapsedMilliseconds}");
    }
}

public static void Work() {}

public static void Test(){
    using new FuncTime();
    Work();
}

**Comment by jnm2** [comment, marked as answer]
One example is of an expression form of a `using` statement. This is a narrower duplicate of https://github.com/dotnet/csharplang/discussions/2894, if `new` is required, and an exact duplicate if not. Slightly related, an expression form just for `out` parameters was discussed in https://github.com/dotnet/csharplang/discussions/3798.

The statement form looks like it is covered by the championed issue https://github.com/dotnet/csharplang/issues/2235. The syntax you propose has an issue brought up there that `using A.B;` could be either a using directive or a top-level using statement. But `using _ = new FuncTime();` or `using var _ = new FuncTime();` might be the syntax.
